### ISSUE TITLE
[Semantics IOS] Accessibility Inspector does not read widgets past the ScrollableView

### ISSUE THREAD

**Issue by charlesburks** [flutter/flutter#97812]
## Details:
If a Flutter app has two or more scrollable horizontal list, the macOS Accessibility Inspector breaks. If the second list has children that cause the list to scroll past the current view, the screen reader stops, gets stuck, and does not read through the list.

## Evidence: 
https://drive.google.com/file/d/1RPj8VyFzUMZYdnDBCRq7oB2cYM3eUVM9/view?usp=sharing

## Code:
Below is a basic code block from a widget:

```dart
Widget getScrollExample(int listSize, double width) {
    return SizedBox(
      width: width,
      height: 100,
      child: ListView.builder(
          shrinkWrap: true,
          scrollDirection: Axis.horizontal,
          itemCount: listSize,
          itemBuilder: (_, i) {
            return Row(
              children: [
                Container(
                  color: Colors.blue,
                  height: 100,
                  width: 100,
                  child: Text(
                    i.toString(),
                    semanticsLabel: 'Box $i',
                  ),
                ),
                const SizedBox(
                  width: 10,
                  height: 10,
                )
              ],
            );
          }),
    );
  }

  @override
  Widget build(BuildContext context) {
    final double width = MediaQuery.of(context).size.width;
    return Scaffold(
        body: CustomScrollView(slivers: [
      SliverFillRemaining(
        hasScrollBody: false,
        child: Column(
          children: [
            const SizedBox(
              width: 10,
              height: 100,
            ),
            getScrollExample(6, width),
            const SizedBox(
              width: 10,
              height: 10,
            ),
            getScrollExample(6, width),
          ],
        ),
      ),
    ]));
  }
```

**Comment by maheshmnj** [comment]
Hi @charlesburks, Thanks for filing the issue,

Are you referring to this issue where it stops reading after the first list and doesn't switch to the second list?

https://user-images.githubusercontent.com/31410839/152780567-3fcecde8-fdff-4ccd-b630-357a4ae6405f.mov

**Comment by charlesburks** [comment]
@maheshmnj correct, the second list is never accessed by the screen reader.

**Comment by maheshmnj** [comment]
Thanks for the clarification, I am able to reproduce the issue on both the stable and the master channel. Accessibility inspector doesn't read past the ScrollableView to any other widget.


*Note: unmute the video for audio visual output*

https://user-images.githubusercontent.com/31410839/152933316-90d922ee-e83a-46c6-9fd8-076372985de3.mov


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Widget getScrollExample(int listSize, double width) {
    return SizedBox(
      width: width,
      height: 100,
      child: ListView.builder(
          shrinkWrap: true,
          scrollDirection: Axis.horizontal,
          itemCount: listSize,
          itemBuilder: (_, i) {
            return Row(
              children: [
                Container(
                  color: Colors.blue,
                  height: 100,
                  width: 100,
                  child: Text(
                    i.toString(),
                    semanticsLabel: 'Box $i',
                  ),
                ),
                const SizedBox(
                  width: 10,
                  height: 10,
                )
              ],
            );
          }),
    );
  }

  @override
  Widget build(BuildContext context) {
    final double width = MediaQuery.of(context).size.width;
    return Scaffold(
        appBar: AppBar(
            title: const Text('Accessibility example'),
            actions: [IconButton(onPressed: () {}, icon: Icon(Icons.add))]),
        body: CustomScrollView(slivers: [
          SliverFillRemaining(
            hasScrollBody: false,
            child: Column(
              children: [
                const SizedBox(
                  width: 10,
                  height: 100,
                ),
                getScrollExample(6, width),
                const SizedBox(
                  width: 10,
                  height: 10,
                ),
                getScrollExample(6, width),
                ElevatedButton(onPressed: () {}, child: Text('Press me')),
                const Placeholder(
                  fallbackHeight: 200,
                  fallbackWidth: 300,
                  color: Colors.red,
                )
              ],
            ),
          ),
        ]));
  }
}

```
</details>


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.10.0, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.10.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5f105a6ca7 (6 days ago), 2022-02-01 14:15:42 -0800
    • Engine revision 776efd2034
    • Dart version 2.16.0
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 14F40EC6-3B73-469A-A1C0-8CF0B152290F • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.1 21C52
      darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 2.11.0-0.0.pre.452, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.11.0-0.0.pre.452 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ba01ec8faa (2 days ago), 2022-02-05 22:05:18 -0500
    • Engine revision 0712096899
    • Dart version 2.17.0 (build 2.17.0-85.0.dev)
    • DevTools version 2.10.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 14F40EC6-3B73-469A-A1C0-8CF0B152290F • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.1 21C52
      darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by goderbauer** [comment]
Just to double-check: This is only an issue with the inspector tool? The UI works as expected with real voiceover?

/cc @chunhtai

**Comment by charlesburks** [comment]
@goderbauer  This is not an issue with the Inspector tool, the reader fails to read anything past the scrollable widget.

**Comment by charlesburks** [comment]
@timsneath @goderbauer
We found another breaking Accessibly issue #99300

**Comment by chunhtai** [comment]
seems to be a duplicate of https://github.com/flutter/flutter/issues/99300. and should probably be fixed at latest master. Closing this issue
